src/events/processSignatureRequests.js

```diff
@@ -46,20 +99,40 @@ function processSignatureRequestsBuilder(config) {
 
         let gasEstimate
         try {
-          gasEstimate = await homeBridge.methods
-            .submitSignature(signature.signature, message)
-            .estimateGas({ from: VALIDATOR_ADDRESS })
+          gasEstimate = await estimateGas(
+            web3Home,
+            homeBridge,
+            signature.signature,
+            message,
+            VALIDATOR_ADDRESS
+          )
+          logger.info(`gasEstimate: ${gasEstimate}`)
         } catch (e) {
-          if (e.message.includes('Invalid JSON RPC response')) {
+          if (e instanceof HttpListProviderError) {
             throw new Error(
-              `RPC Connection Error: submitSignature Gas Estimate cannot be obtained.`
+              'RPC Connection Error: submitSignature Gas Estimate cannot be obtained.'
+            )
+          } else if (e instanceof InvalidValidatorError) {
+            logger.warn({ address: VALIDATOR_ADDRESS }, 'Invalid validator')
+            throw new Error('Current address does not correspond to a validator')
+          } else if (e instanceof AlreadySignedError) {
+            logger.info(
+              { eventTransactionHash: signatureRequest.transactionHash },
+              `Already signed signatureRequest ${signatureRequest.transactionHash}`
             )
+            return
+          } else if (e instanceof AlreadyProcessedError) {
+            logger.info(
+              { eventTransactionHash: signatureRequest.transactionHash },
+              `signatureRequest ${
+                signatureRequest.transactionHash
+              } was already processed by other validators`
+            )
+            return
+          } else {
+            logger.error(e, 'Unknown error while processing transaction')
+            throw e
```

Comment:
The logic here is: if we `return`, then this transaction is ignored (this is done when the transaction was already signed or if it was processed by other validators). If we `throw`, the transaction will be retried later (this is done for connection errors, unknown errors and when the current address is not a validator).

I'm not sure about retrying when the address is not a validator, though. Maybe we should just `process.exit(1)` and be done with it?